c/wrapper.c

```
@@ -246,31 +250,40 @@ uint16_t jl_unbox_complex16_im(jl_value_t *v) {
+  properties_value = jl_call1(propertynames, v);
+  jl_array_t *properties = (jl_array_t *)properties_value;
   size_t len = jl_array_len(properties);
   const char **names = (const char **)malloc(len * sizeof(char *));
   for (size_t i = 0; i < len; i++) {
-#if JL_VERSION_AT_LEAST(1, 11)
-    jl_value_t *name = jl_array_data(properties, jl_value_t);
-#else
-    jl_value_t *name = jl_array_data(properties);
-#endif
+    jl_value_t *name = jl_array_ptr_ref(properties, i);
     names[i] = jl_symbol_name((jl_sym_t *)name);
```

Cppcheck (2.20.0)

[warning] If memory allocation fails, then there is a possible null pointer dereference

(nullPointerOutOfMemory)